connexion/operations/secure.py

```diff
@@ -76,72 +76,84 @@ def security_decorator(self):
             if not security_req:
                 auth_funcs.append(self._api.security_handler_factory.verify_none())
                 continue
-            elif len(security_req) > 1:
-                logger.warning("... More than one security scheme in security requirement defined. "
-                               "**DENYING ALL REQUESTS**", extra=vars(self))
-                return self._api.security_handler_factory.security_deny
-
-            scheme_name, scopes = next(iter(security_req.items()))
-            security_scheme = self.security_schemes[scheme_name]
-
-            if security_scheme['type'] == 'oauth2':
-                required_scopes = scopes
-                token_info_func = self._api.security_handler_factory.get_tokeninfo_func(security_scheme)
-                scope_validate_func = self._api.security_handler_factory.get_scope_validate_func(security_scheme)
-                if not token_info_func:
-                    logger.warning("... x-tokenInfoFunc missing", extra=vars(self))
-                    continue
-
-                auth_funcs.append(self._api.security_handler_factory.verify_oauth(token_info_func, scope_validate_func))
-
-            # Swagger 2.0
-            elif security_scheme['type'] == 'basic':
-                basic_info_func = self._api.security_handler_factory.get_basicinfo_func(security_scheme)
-                if not basic_info_func:
-                    logger.warning("... x-basicInfoFunc missing", extra=vars(self))
-                    continue
-
-                auth_funcs.append(self._api.security_handler_factory.verify_basic(basic_info_func))
-
-            # OpenAPI 3.0.0
-            elif security_scheme['type'] == 'http':
-                scheme = security_scheme['scheme'].lower()
-                if scheme == 'basic':
+
+            sec_req_funcs = {}
+            oauth = False
+            for scheme_name, scopes in security_req.items():
+                security_scheme = self.security_schemes[scheme_name]
+
+                if security_scheme['type'] == 'oauth2':
+                    if oauth:
+                        logger.warning("... multiple OAuth2 security schemes in AND fashion not supported", extra=vars(self))
```

Comment:
Should we `return self._api.security_handler_factory.security_deny` here as was previously the case for multiple security schemes?

Reply:
We could do that. Then it's a bit inconsistent with the behavior when a `x-tokenInfoFunc` or `x-apikeyInfoFunc` is missing (then you can't use the corresponding security scheme to authenticate, but you can still use another authentication option if there are multiple).

If we do want to do something like the `security_deny`, I would perhaps even raise an error such that the server doesn't even start, instead of starting it but refusing any requests for that endpoint?

Reply:
On a second look, I think the current implementation is indeed more consistent. When encountering an issue in the definition of a security scheme, we should invalidate all security which is defined in an AND fashion with this scheme, but any (set of) schemes defined in an OR fashion can still be used independently.

Raising an error for any issue in the definition could be a good alternative, but it's quite a big change compared to current Connexion behavior. If we wanted to do this, I think we should at least provide a setting for it on the application level. For now I would leave it as is, but could be a nice addition for the future.